examples/nlp/lasertagger/lasertagger_main.py

```diff
@@ -0,0 +1,368 @@
+    )
+    parser_infer.add_argument(
+        "--test_file_raw", type=str, help="The path to test tsv file",
+    )
+    parser_infer.add_argument(
+        "--test_file_preprocessed", type=str, help="The path to preprocessed test pkl file",
+    )
+    parser_infer.add_argument(
+        "--label_map_file",
+        type=str,
+        help="Path to the label map file. Either a JSON file ending with '.json', \
+				that maps each possible tag to an ID, or a text file that \
+				has one tag per line.",
+    )
+    parser_infer.add_argument(
+        '--pretrained_model_name',
+        default='bert-base-cased',
+        type=str,
+        help='Name of the pre-trained model',
+        choices=nemo_nlp.nm.trainables.get_pretrained_lm_models_list(),
+    )
+    parser_infer.add_argument(
+        '--bert_checkpoint', default='bert-base-cased', type=str, help='Path to the pre-trained model checkpoint'
+    )
+    parser_infer.add_argument(
+        "--model_config_file", default=None, type=str, help="Path to LaserTagger config file in json format"
+    )
+    parser_infer.add_argument("--vocab_file", default=None, help="Path to the vocab file.")
+    parser_infer.add_argument("--max_seq_length", default=128, type=int)
+
+    return parser.parse_args()
+
+
+def _calculate_steps(num_examples, batch_size, num_epochs, warmup_proportion=0):
+    """Calculates the number of steps.
+
+  Args:
+	num_examples: Number of examples in the dataset.
+	batch_size: Batch size.
+	num_epochs: How many times we should go through the dataset.
+	warmup_proportion: Proportion of warmup steps.
+
+  Returns:
+	Tuple (number of steps, number of warmup steps).
+  """
+    steps = int(num_examples / batch_size * num_epochs)
+    warmup_steps = int(warmup_proportion * steps)
+    return steps, warmup_steps
+
+
+if __name__ == "__main__":
+
+    args = parse_args()
+
+    if args.command == 'train':
+        train_examples, num_train_examples = torch.load(args.train_file)
+        if args.eval_file:
+            eval_examples, num_eval_examples, eval_special_tokens = torch.load(args.eval_file)
+    test_examples, num_test_examples, test_special_tokens = torch.load(args.test_file_preprocessed)
+
+    label_map = utils.read_label_map(args.label_map_file)
+    num_tags = len(label_map)
+
+    config = None
+    with open(args.model_config_file, "r", encoding="utf-8") as reader:
+        text = reader.read()
+        config = json.loads(text)
+
+    nf = nemo_core.NeuralModuleFactory(
+        backend=nemo_core.Backend.PyTorch,
+        local_rank=args.local_rank,
+        optimization_level=args.amp_opt_level,
+        log_dir=args.work_dir,
+        create_tb_writer=True,
+        files_to_copy=[__file__],
+        add_time_to_log_dir=False,
+    )
+
+    encoder = nemo_nlp.nm.trainables.huggingface.BERT(pretrained_model_name=args.pretrained_model_name)
+
+    tokenizer = nemo_nlp.data.NemoBertTokenizer(pretrained_model=args.pretrained_model_name)
```

Comment:
so it looks like you are creating a tokenizer from HuggingFace's cloud, then why do you need to keep vocab files in the data folder?

Reply:
Agree, this can be removed.